pkg/ring/model.go

```diff
@@ -182,12 +182,20 @@ func (d *Desc) TokensFor(id string) (tokens, other []uint32) {
 
 // IsHealthy checks whether the ingester appears to be alive and heartbeating
 func (i *IngesterDesc) IsHealthy(op Operation, heartbeatTimeout time.Duration) bool {
-	if op == Write && i.State != ACTIVE {
-		return false
-	} else if op == Read && i.State == JOINING {
-		return false
+	healthy := false
+
+	switch op {
+	case Write:
+		healthy = i.State == ACTIVE
+
+	case Read:
+		healthy = i.State == ACTIVE || i.State == LEAVING || i.State == PENDING
+
+	case Reporting:
+		healthy = true
```

Comment:
If we want 1:1 mapping of the functionality, then this should be `default` case.

Reply:
Then it would never be set to false.

Reply:
Ah, sorry, forget my previous comment. It would of course. But there is no other `op`, so having Reporting case here covers all cases.

Reply:
How? It would fall back to `default` only if the `op` was not `Write` or `Read`.

Reply:
Yeah, it looks fine. There is no hard requirement from Go that it should be only those 3 values, but for our use case it should be fine I guess.